File: solutions/Longest-Common-Prefix/Longest-Common-Prefix.cpp

class Solution {
public:
    string longestCommonPrefix(vector<string> &strs) {
        if(strs.size()==0)
            return "";
        int minlen=INT_MAX;
        string mark="";
        for(int i=0;i<strs.size();i++)
            if(strs[i].length()<minlen)
            {
                mark=strs[i];
                minlen=strs[i].length();
            }
        string ans="";
        for(int i=0;i<mark.length();i++)
        {
            for(int j=0;j<strs.size();j++)
                if(strs[j][i]!=mark[i])
                    return ans;
            ans+=mark[i];
        }
        return ans;
    }
};


 

File: solutions/Longest-Common-Prefix/Longest-Common-Prefix.py


class Solution:
    # @return a string
    def longestCommonPrefix(self, strs):
        if len(strs)==0:
            return ""
        minlen=10000
        mark=0
        for i in range(len(strs)):
            if len(strs[i])<minlen:
                minlen=len(strs[i])
                mark=i
        ans=""
        for i in range(len(strs[mark])):
            for j in range(len(strs)):
                if strs[j][i]!=strs[mark][i]:
                    return ans
            ans+=strs[mark][i]
        return ans
 